lib/puppet/provider/snmp_notification/cisco_ios.rb

```diff
@@ -44,10 +44,31 @@ def self.commands_from_instance(property_hash)
       commands_array
     end
 
-    def get(context, _names = nil)
+    def get(context, names = [])
       output = context.transport.run_command_enable_mode(PuppetX::CiscoIOS::Utility.get_values(commands_hash))
       return [] if output.nil?
       return_value = Puppet::Provider::SnmpNotification::CiscoIos.instances_from_cli(output)
+      unless names.empty?
+        # as the `enable` property is like an
+        # ensure except it does not get returned
+        # if the trap is not enabled
+        # we create our own return value
+        # when the return values from system
+        # does not contain the name requested, i.e.
+        # in an apply - this will allow the provider to be
+        # idempotent on both a `true` and `false` snmp_notification
+        # resource
+        names.each do |name|
+          return_value << { name: name, enable: false } if return_value.empty?
+          return_value.select do |rval|
+            next unless rval[:name] != name
+            return_value << {
+              name: name,
+              enable: false,
+            }
+          end
+        end
+      end
```

Comment:
Rewrite this to create a lookup-hash from `return_value` by name and then replace the inner `return_value.select` with a simple "lookup or synthesise".

Reply:
Something like `return_value.select { |rVal| rVal[:name] == name ? rVal : return_value << { name: name, enable: false } }`?

Reply:
more like
```
instances_by_name = Hash[return_value.select { |v| [v[:name], v] } ]
names.each { |n| instances_by_name[n] ||= { name: n, enable: false} }
```